### PR TITLE
Node/Snap/Layers Test Coverage and Bug Fixes

### PR DESCRIPTION
**Pull request by vertexmc**
- Bug Fixes:
    - [x] Tag time ranges on nodes would never update after being initially set
    - [x] Tag times would be lost on bulk addNodes
    - [x] Snap permissions cache would persist after snap's user changed
    - [x] Cleaned up imports and exception in files edited, then cleaned up other code that was broken as a result
- Test coverage for `synapse/lib/{node, snap}`

### CODE REVIEW

File: synapse/tests/test_lib_node.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                snap.strict = False
+                self.false(node.pop('nope'))
+                snap.strict = True
+
+                with self.getTestDir() as dirn:
+                    with s_auth.Auth(dirn) as auth:
+                        user = auth.addUser('hatguy')
+                        snap.setUser(user)
+
+                        self.raises(s_exc.AuthDeny, node.set, 'tick', 1)
```
Comment (vertexmc) [inline synapse/tests/test_lib_node.py]: @invisig0th - Should this raise AuthDeny? It currently does. The user has no perms configured

File: synapse/tests/test_lib_node.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+                self.false(node.set('tick', 12345))
+                self.raises(s_exc.NoSuchProp, node.set, 'notreal', 12345)
+
+                with self.getTestDir() as dirn:
+                    with s_auth.Auth(dirn) as auth:
+                        user = auth.addUser('hatguy2')
+                        snap.setUser(user)
+
+                        self.raises(s_exc.AuthDeny, node.set, 'tick', 1)
```
[the remark this answers is not in the document]
Comment (vertexmc) [reply synapse/tests/test_lib_node.py]: Fixed

File: synapse/tests/test_lib_node.py (hunk 1 above)
Comment (vertexmc) [reply synapse/tests/test_lib_node.py]: Fixed

File: synapse/lib/snap.py [hunk 3]
```diff
@@ -39,6 +37,8 @@ def __init__(self, core, layers, write=False):
         self.strict = True
         self.elevated = False
 
+        self.permcache = s_cache.FixedCache({}, size=1000)
```
Comment (vertexmc) [inline synapse/lib/snap.py]: Setting `callback` to `{}` is probably the wrong thing to do in the case where these is no user. Is there something better I could do here?

File: synapse/lib/snap.py [hunk 4]
```diff
@@ -101,17 +101,22 @@ def getOffset(self, iden, offs):
         return self.xact.getOffset(iden, offs)
 
     def setUser(self, user):
+
         self.user = user
+        self.permcache.clear()
+        self.permcache.callback = {}
```
Comment (vertexmc) [inline synapse/lib/snap.py]: Same `callback` comment as above

File: synapse/lib/snap.py [hunk 5]
```diff
@@ -174,8 +179,6 @@ def _getNodesByFormTag(self, name, tag, valu=None, cmpr='='):
         fenc = form.name.encode('utf8') + b'\x00'
         tenc = b'#' + tag.encode('utf8') + b'\x00'
 
-        pref = fenc + tenc
```
Comment (vertexmc) [inline synapse/lib/snap.py]: unused

File: synapse/lib/snap.py [hunk 6]
```diff
@@ -282,19 +286,14 @@ def _addNodeFnib(self, fnib, props=None):
         if props is None:
             props = {}
 
-        # store an original copy of the props for the splice.
-        orig = props.copy()
```
Comment (vertexmc) [inline synapse/lib/snap.py]: unused

File: synapse/lib/snap.py [hunk 7]
```diff
@@ -282,19 +286,14 @@ def _addNodeFnib(self, fnib, props=None):
...
-
         sops = []
 
-        init = False
```
Comment (vertexmc) [inline synapse/lib/snap.py]: unused

File: synapse/tests/test_cortex.py [hunk 8]
```diff
@@ -1,26 +1,22 @@
 import synapse.exc as s_exc
 import synapse.common as s_common
+import synapse.tests.common as s_t_common
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: We've been doing this as `import synapse.tests.common as s_test` in other code

Comment (vertexmc) [reply synapse/tests/test_cortex.py]: Fixing :-)

File: synapse/lib/snap.py (hunk 4 above)
Comment (vertexmc) [reply synapse/lib/snap.py]: Comment was removed. Should we use something else for the callback?

File: synapse/common.py [hunk 9]
```diff
@@ -1,3 +1,4 @@
+# stdlib
```
Comment (jnwatson) [inline synapse/common.py]: I'm not a big fan of the include section comments. A blank line is sufficient.

**Pull request merged**